test/test_distributions.py

```diff
@@ -215,7 +230,8 @@ def _check_sampler_sampler(self, torch_dist, ref_dist, message, multivariate=Fal
             axis /= np.linalg.norm(axis)
             torch_samples = np.dot(torch_samples, axis)
             ref_samples = np.dot(ref_samples, axis)
-        samples = [(x, +1) for x in torch_samples] + [(x, -1) for x in ref_samples]
+	samples = [(x, +1) for x in torch_samples] + [(x, -1) for x in ref_samples]
+        shuffle(samples) # necessary to prevent stable sort from making uneven bins for discrete
```

Comment:
As a quick explanation: if a discrete distribution has a small support, than the bins can be uneven because the above line just concats the lists, and so one bin can have all the same values as the next, but be filled with samples from different distributions. This shuffle is a quick way to prevent this from happening, and is the right way to do this test.